src/ape/api/networks.py

```diff
@@ -214,7 +213,11 @@ def encode_transaction(
         ...
 
     @abstractmethod
-    def decode_event(self, abi: EventABI, receipt: "ReceiptAPI") -> "ContractLog":
+    def encode_log_filter(self, abi: EventABI, **filter_args) -> Dict:
```

Comment:
I think this function would be `Web3` provider-specific, probably best off living there

Reply:
Ok, I can put it there, but it needs to do some of the stuff that `ape_ethereum` also needs to do... 

I am struggling to find a clean way to share stuff and some code may have to be duplicated